Sources/PromotedCore/MetricsLogger+Helpers.swift

```diff
@@ -1,189 +1,124 @@
   }
+}
 
-  /// Logs an action to share the given content.
-  @objc func logShareAction(content: Content) {
-    logAction(name: "share",
-              type: .share,
-              contentID: content.contentID,
-              insertionID: content.insertionID)
+// MARK: - Action Logging Helper Methods (ObjC)
+public extension MetricsLogger {
+  /// Logs an action with given type and content.
+  @objc(logActionWithType:content:)
+  func _objcLogAction(type: ActionType, content: Content?) {
+    logAction(
+      name: type.description,
+      type: type,
+      contentID: content?.contentID,
+      insertionID: content?.insertionID
+    )
   }
   
-  /// Logs an action to like the given content.
-  @objc func logLikeAction(content: Content) {
-    logAction(name: "like",
-              type: .like,
-              contentID: content.contentID,
-              insertionID: content.insertionID)
-  }
-
-  /// Logs an action to unlike the given content.
-  @objc func logUnlikeAction(content: Content) {
-    logAction(name: "unlike",
-              type: .unlike,
-              contentID: content.contentID,
-              insertionID: content.insertionID)
-  }
-
-  /// Logs an action to comment on the given content.
-  @objc func logCommentAction(content: Content) {
-    logAction(name: "comment",
-              type: .comment,
-              contentID: content.contentID,
-              insertionID: content.insertionID)
-  }
-
-  /// Logs an action to make an offer on the given item.
-  @objc func logMakeOfferAction(item: Item) {
-    logAction(name: "make-offer",
-              type: .makeOffer,
-              contentID: item.contentID,
-              insertionID: item.insertionID)
-  }
-
-  /// Logs an action to ask a question about the given content.
-  @objc func logAskQuestionAction(content: Content) {
-    logAction(name: "ask-question",
-              type: .askQuestion,
-              contentID: content.contentID,
-              insertionID: content.insertionID)
-  }
-
-  /// Logs an action to answer a question about the given content.
-  @objc func logAnswerQuestionAction(content: Content) {
-    logAction(name: "answer-question",
-              type: .answerQuestion,
-              contentID: content.contentID,
-              insertionID: content.insertionID)
-  }
-
-  /// Logs an action to sign in to an account.
-  @objc func logCompleteSignInAction() {
-    logAction(name: "sign-in",
-              type: .completeSignIn,
-              contentID: nil,
-              insertionID: nil)
-  }
-
-  /// Logs an action to sign up for an account.
-  @objc func logCompleteSignUpAction() {
-    logAction(name: "sign-up",
-              type: .completeSignUp,
-              contentID: nil,
-              insertionID: nil)
+  /// Logs an action with given type, content, and name.
+  @objc(logActionWithType:content:name:)
+  func _objcLogAction(type: ActionType, content: Content?, name: String?) {
+    logAction(
+      name: name ?? type.description,
```

Comment:
Unrelated to PR - you don't have to fill in name.  If we want a default name, we can figure that out in a later step.

Reply:
Do you prefer the client leaves it blank? That's easy to do. I don't have a strong preference on the client side.